FirebaseAuth/Sources/Swift/User/User.swift

```diff
@@ -2112,18 +2123,20 @@ extension User: NSSecureCoding {}
       of: NSString.self,
       forKey: kDisplayNameCodingKey
     ) as? String
-    let photoURL = coder.decodeObject(forKey: kPhotoURLCodingKey) as? URL
+    let photoURL = coder.decodeObject(of: NSURL.self, forKey: kPhotoURLCodingKey) as? URL
     let email = coder.decodeObject(of: NSString.self, forKey: kEmailCodingKey) as? String
     let phoneNumber = coder.decodeObject(
       of: NSString.self,
       forKey: kPhoneNumberCodingKey
     ) as? String
     let emailVerified = coder.decodeBool(forKey: kEmailVerifiedCodingKey)
-    let providerData = coder.decodeObject(forKey: kProviderDataKey) as? [String: UserInfoImpl]
-    let metadata = coder.decodeObject(forKey: kMetadataCodingKey) as? UserMetadata
+    let classes = [NSDictionary.self, NSString.self, UserInfoImpl.self]
+    let providerData = coder.decodeObject(of: classes, forKey: kProviderDataKey)
```

Comment:
This one was especially hard to debug, since I started by investigating why metadata wasn't getting populated in the quickstart after decoding.

It turns out that even theough the Metadata decode was fine, every decode after the failing providerData decode failed silently with nils.